### ISSUE TITLE
after compress the apng file, YYimage decode can not get the frames count of the apng file.

### ISSUE THREAD

**Issue by y500** [shssoichiro/oxipng#625]
when using this tool on an APNG file, a problem was discovered.

before compressing, the YYImage decoder can decode it perfectly as expected, after compressed, the YYImage decoder can not get the frame count of the file and can not animate the apng image.

I tried to use ffmpeg to compare two files:

**origin file**
```
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
Input #0, apng, from 'gt_apng_before.png':
  Duration: N/A, bitrate: N/A
  Stream #0:0: Video: apng, pal8(pc), 60x60, 10.42 fps, 24 tbr, 100k tbn, 100k tbc
```

**compressed file**
```
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
[apng @ 0x7fb323808200] Could not find codec parameters for stream 0 (Video: apng, none(pc), 60x60): unspecified pixel format
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
Input #0, apng, from 'gt_apng_after.png':
  Duration: N/A, bitrate: N/A
  Stream #0:0: Video: apng, none(pc), 60x60, 100k tbr, 100k tbn, 100k tbc
At least one output file must be specified
```

the "pal8(pc)" is gone and ffmpeg can't decode it too.

And on the Mac, the two files can preview normally.

**Comment by TPS** [comment]
As a test, does running https://github.com/jindongyi011039/apngopt against your original APNG, & then ruuning the output through the above give the same result?

**Comment by andrews05** [comment]
Looks like some chunks are in a slightly different order.
Before:
```
  chunk IHDR at offset 0x0000c, length 13
    320 x 320 image, 8-bit palette, non-interlaced
  chunk acTL at offset 0x00025, length 8
    unknown private, ancillary, unsafe-to-copy chunk
  chunk PLTE at offset 0x00039, length 762: 254 palette entries
  chunk tRNS at offset 0x0033f, length 1: 1 transparency entry
  chunk fcTL at offset 0x0034c, length 26
    unknown private, ancillary, unsafe-to-copy chunk
  chunk IDAT at offset 0x00372, length 12143
    zlib: deflated, 32K window, maximum compression
  ...
```
After:
```
  chunk IHDR at offset 0x0000c, length 13
    320 x 320 image, 8-bit palette, non-interlaced
  chunk acTL at offset 0x00025, length 8
    unknown private, ancillary, unsafe-to-copy chunk
  chunk fcTL at offset 0x00039, length 26
    unknown private, ancillary, unsafe-to-copy chunk
  chunk PLTE at offset 0x0005f, length 762: 254 palette entries
  chunk tRNS at offset 0x00365, length 1: 1 transparency entry
  chunk IDAT at offset 0x00372, length 11577
    zlib: deflated, 32K window, maximum compression
  ...
``` 
The apng spec makes no mention of required order of fcTL relative to PLTE or tRNS, so it's a bit unexpected that this would cause some programs to fail. I'm sure we can get a fix out though.

**Comment by y500** [comment]
thanks a lot and waiting for the release.

**Comment by andrews05** [comment]
@y500 This is available now in v9.1.2. If you have any existing images that were affected by this, you should be able to "fix" them by running oxipng with `--force`.
